pkg/authenticator/api-key.go

```diff
@@ -54,20 +54,26 @@ func (aa *APIKeyAuthenticator) GetToken(freshTokenRequired bool) (string, uint64
 		}
 	}
 
-	aa.logger.Info("Fetching fresh token")
-	tokenResponse, err := aa.authenticator.RequestToken()
+	var tokenResponse *core.IamTokenServerResponse
+	err = retry(aa.logger, func() error {
+		tokenResponse, err = aa.authenticator.RequestToken()
+		return err
+	})
+
 	if err != nil {
-		aa.logger.Error("Error fetching fresh token", zap.Error(err))
-		// If the cluster cannot access private iam endpoint, hence returns timeout error, switch to public IAM endpoint.
-		if !isTimeout(err) {
+		// Reset the IAM URL to public, if it is private
+		if !resetIAMURL(aa) {
```

Comment:
when reset is required ?

Reply:
When timeout error still exists using private endpoint.